src/transformers/modeling_t5.py

```diff
@@ -916,3 +916,23 @@ def forward(self, **kwargs):
             ) + decoder_outputs  # TODO(thom): Add z_loss https://github.com/tensorflow/mesh/blob/fa19d69eafc9a482aff0b59ddd96b025c0cb207d/mesh_tensorflow/layers.py#L666
 
         return decoder_outputs + encoder_outputs
+
+    def prepare_inputs_for_generation(self, input_ids, past, encoder_inputs, attention_mask):
```

Comment:
@craffel - This function overrides the default `prepare_inputs_for_generation` in `modeling_utils.py` and "sorts the inputs correctly for T5. Did not handle the input_embeds and `decoder_hidden_states` there yet...